src/metadata/gas.rs

```diff
@@ -100,17 +102,19 @@ impl GasMetadata {
         Some(
             self.gas_info.function_costs[func]
                 .iter()
-                .map(|(token_type, val)| val.into_or_panic::<usize>() * token_gas_cost(*token_type))
+                .map(|(token_type, val)| (*val as usize) * token_gas_cost(*token_type))
```

Comment:
Using `as` can also fail, but it will do so silently. I think it's better to return an error in those cases (a NativeAssert)

Reply:
The problem with that is that the `initial_required_gas` is used by `get_initial_available_gas` which already returns an error. So I don't think that returning a NativeAssert would solve this situation.